src/app/applications/iot-devices/iot-device-detail/downlink/downlink-tables/downlink-tables.component.ts

```diff
@@ -0,0 +1,136 @@
+import { DownlinkDialogComponent } from "../downlink-dialog/downlink-dialog.component";
+import { Downlink } from "../downlink.model";
+import { DownlinkQueueDto } from "../downlink-queue-dto";
+
+@Component({
+  selector: "app-downlink-tables",
+  templateUrl: "./downlink-tables.component.html",
+  styleUrls: ["./downlink-tables.component.scss"],
+})
+export class DownlinkTablesComponent implements OnInit {
+  @Input() device: IotDevice;
+  @Input() errorMessages: string[];
+  @Input() downlinkQueue: DownlinkQueueDto[];
+  public downlinkHistoryQueue: DownlinkQueueDto[];
+  public isLoadingResults = false;
+  public displayedColumns: string[] = ["createdAt", "status", "payload", "port"];
+  public displayedColumnsHistory: string[] = [
+    "createdAt",
+    "sendAt",
+    "acknowledgedAt",
+    "status",
+    "fCntDown",
+    "acknowlegded",
+    "payload",
+    "port",
+  ];
+
+  constructor(
+    private snackBar: MatSnackBar,
+    private translate: TranslateService,
+    private downlinkService: DownlinkService,
+    public dialog: MatDialog,
+    private errorMessageService: ErrorMessageService
+  ) {}
+
+  ngOnInit(): void {
+    this.errorMessages = [];
+
+    this.getDownlinksQueue();
+    this.getHistoricalDownlinksQueue();
+  }
+
+  getDownlinksQueue() {
+    this.isLoadingResults = true;
+    this.downlinkService.getDownlinkQueue(this.device.id).subscribe(
+      (response: DownlinkQueueDto[]) => {
+        this.downlinkQueue = response;
+        this.isLoadingResults = false;
+      },
+      error => {
+        this.handleError(error);
+        this.isLoadingResults = false;
+      }
+    );
+  }
+
+  getHistoricalDownlinksQueue() {
+    this.isLoadingResults = true;
+    this.downlinkService.getHistoricalDownlinkQueue(this.device.id).subscribe(
+      (response: DownlinkQueueDto[]) => {
+        this.downlinkHistoryQueue = response;
+        this.isLoadingResults = false;
+      },
+      error => {
+        this.handleError(error);
+        this.isLoadingResults = false;
+      }
+    );
+  }
+
+  clickReload() {
+    this.getDownlinksQueue();
+    this.getHistoricalDownlinksQueue();
+  }
+
+  clickFlushQueue() {
+    this.openDownlinkDialog();
+  }
+
+  private handleError(error: HttpErrorResponse) {
+    this.errorMessages = this.errorMessageService.handleErrorMessage(error);
+  }
+
+  openDownlinkDialog() {
+    const dialog = this.dialog.open(DownlinkDialogComponent, {});
+
+    dialog.afterClosed().subscribe(result => {
+      if (result === true) {
+        this.downlinkService.flushQueue(this.device.id).subscribe(
+          response => {
+            this.snackBar.open("Kø ryddet", "OK", {
+              duration: 10000,
+            });
+            this.getDownlinksQueue();
+          },
+          error => {
+            console.log(error);
+            this.handleError(error);
+          }
+        );
+      }
+    });
+  }
+
+  checkForLoRaWAN(device: IotDevice) {
+    return device.type === DeviceType.LORAWAN;
```

Comment:
Du bruger den her et par steder, men hele komponentet er ikke rendered hvis det er lorawan, så det bør være redundant de to steder du har den her